lib/fog/storage/google_json/requests/put_object.rb

```diff
@@ -52,12 +52,14 @@ def put_object(bucket_name,
               options[:content_type] = "text/plain"
             elsif data.is_a?(::File)
               options[:content_type] = Fog::Storage.parse_data(data)[:headers]["Content-Type"]
-            elsif data.respond_to?(:content_type) && data.respond_to?(:path)
-              options[:content_type] = data.content_type
-              data = data.path
             end
           end
 
+          if data.respond_to?(:content_type) && data.respond_to?(:path)
+            options[:content_type] = data.content_type
```

Comment:
Random thought, should we not overwrite content_type in this case? 

Also, mind putting a comment right above this `if` saying that this is for Paperclip? Just so it doesn't get erased.

Reply:
> should we not overwrite content_type in this case?

I’m not sure I’m following. Do you mean we should prefer `options[:content_type]` over `data.content_type`?

> mind putting a comment right above this `if` saying that this is for Paperclip?

Done 👍

Reply:
@sonicdoe Yeah, I mean, the user in this case is passing in a content type, which kind of implies to me they want that to be the content type, despite whatever the file actually thinks it is. We could do something like `options[:content_type] ||= data.content_type` to make sure there is always a content type, but default to the one the user passes in.

Reply:
You’re right, this does indeed make more sense. It likely won’t matter in practice since this part is probably only used by Paperclip (which always passes the file’s content type).

Anyway, I’ve changed it to `||=` in d814259e2e02ecb93522e755f9efa8c5b7224748 👍